integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciConfigBean.java

```diff
@@ -0,0 +1,347 @@
+     */
+    @ConfiguredOption(key = "auth.tenant-id")
+    Optional<String> authTenantId();
+
+    /**
+     * The OCI user id.
+     * <p>
+     * This configuration property has an effect only when {@code config} is, explicitly or implicitly,
+     * present in the value for the {@link #authStrategies()}.
+     * When it is present, this property must be provided in order to set the
+     * {@linkplain ConfigFileAuthenticationDetailsProvider#getUserId()}.
+     *
+     * @return the OCI user id
+     */
+    @ConfiguredOption(key = "auth.user-id")
+    Optional<String> authUserId();
+
+    /**
+     * The OCI IMDS hostname.
+     * <p>
+     * This configuration property is used to identify the metadata service url.
+     *
+     * @return the OCI IMDS hostname
+     */
+    @ConfiguredOption(value = IMDS_HOSTNAME, key = "imds.hostname")
+    String imdsHostName();
+
+    /**
+     * The OCI IMDS connection timeout in millis. This is used to auto-detect availability.
+     * <p>
+     * This configuration property is used when attempting to connect to the metadata service.
+     *
+     * @return the OCI IMDS connection timeout in millis
+     * @see OciAvailability
+     */
+    @ConfiguredOption(value = "100", key = "imds.timeout.milliseconds")
+    int imdsTimeoutMilliseconds();
+
+    /**
+     * The list of {@link AuthStrategy} names
+     * (excluding {@link AuthStrategy#AUTO}) that
+     * are potentially applicable for use. Here, "potentially applicable or use" means that it is set explicitly by
+     * {@link #authStrategy()}, or else explicitly or implicitly by {@link #authStrategies()}.
+     *
+     * @return the list of potential auth strategies that are applicable
+     */
+    default List<String> potentialAuthStrategies() {
+        String authStrategy = authStrategy().orElse(null);
+        if (authStrategy != null
+                && !TAG_AUTO.equalsIgnoreCase(authStrategy)
+                && !authStrategy.isBlank()) {
+            if (!ALL_STRATEGIES.contains(authStrategy)) {
+                throw new IllegalStateException("Unknown auth strategy: " + authStrategy);
+            }
+
+            return List.of(authStrategy);
+        }
+
+        List<String> result = new ArrayList<>();
+        authStrategies().stream()
+                .map(String::trim)
+                .filter(Predicate.not(String::isBlank))
+                .forEach(s -> {
+                    if (!ALL_STRATEGIES.contains(s) && !TAG_AUTO.equals(s)) {
+                        throw new IllegalStateException("Unknown auth strategy: " + s);
+                    }
+                    result.add(s);
+                });
+        if (result.isEmpty() || result.contains(TAG_AUTO)) {
+            return ALL_STRATEGIES;
+        }
+
+        return result;
+    }
+
+    /**
+     * Determines whether there is sufficient configuration defined in this bean to be used for file-based authentication. This
+     * matches to the {@link AuthStrategy#CONFIG_FILE}.
```

Comment:
I don't understand this.

Reply:
i'm not sure what else I could say to make it clearer.